src/mtool/api/rest/rest_api/perf/system_perf.py

```diff
@@ -65,6 +64,42 @@ def get_agg_volumes_perf(ip, port):
     
     return res_dict
 
+def set_telemetry_properties(properties):
+    props = {}
+    for prop in properties:
+        for field in prop["fields"]:
+            if field["isSet"]:
+                props[field["field"]] = None
+    params = {
+        "param": {
+            "metrics_to_publish": props
+        }
+    }
+    return ibofos.set_telemetry_properties(params)
+
+def get_telemetry_properties():
+    f = open('util/telemetry/fields.json')
+    props = json.loads(f.read())
+    status = False
+    dagent_telemetry_properties = ibofos.get_telemetry_properties()
+    if "result" in dagent_telemetry_properties and \
```

Comment:
If we get empty or non json response from ibofos.get_telemetry_properties() in case of status.code is not 200 then Will below code be able to handle ?

Reply:
I have updated the error handling. Thanks for the review